### DISCUSSION TITLE
allow use of process substitution for config files

### DISCUSSION THREAD

**Discussion by jackkav** [Kong/kong#7344] [Ideas and feature requests]
### Summary
I'd like --config to be able to parse strings as well as files, such that I can start kong in dbless mode with the minimum of fuss.

Current: must create a file in order to work.
```
echo '{"_format_version":"1.1","services":[{"host":"mockbin.com","port":443,"protocol":"https","routes":[{"paths":["/"]}]}],"plugins":[{"name":"rate-limiting","config":{"policy":"local","limit_by":"ip","minute":3}}]}' > $HOME/config.json && docker run -v "$HOME/config.json:/config.json" -e KONG_DATABASE=off -e KONG_DECLARATIVE_CONFIG=/config.json -p 8000:8000 kong:latest
```

Desired: I would like to start kong in dbless mode using a string rather than a file
```
$ docker run -e KONG_DATABASE=off -p 8000:8000 kong:latest kong start --conf  <(echo '{"_format_version":"1.1","services":[{"host":"mockbin.com","port":443,"protocol":"https","routes":[{"paths":["/"]}]}],"plugins":[{"name":"rate-limiting","config":{"policy":"local","limit_by":"ip","minute":3}}]}')

Error: Failed parsing:
/dev/fd/11: No such file or directory
```

**Related issue:**
https://github.com/Kong/kong/issues/6928

**Comment by javierguerragiraldez** [comment]
a bit of context here:  this feature request is to be able to start Kong without creating a config file.  it seems that Kong actually sees the content passed by the process substitution, but rejects because it's a named pipe and not a regular file.

we could investigate if it's possible to support non-regular files without adding too much special case code

**Comment by eskerda** [comment]
Was going to add a feature request to expose the `declarative_config` setting through a `cmd` flag, like `kong start --yml path/to/file.yml`. This is kind of related.

**Comment by jackkav** [comment]
Thanks @guanlan for starting the discussion. Made a POC over here https://github.com/Kong/kong/pull/7379
